### PR TITLE
Support copy to clipboard for code snippets in ballerina usecase pages

### PR DESCRIPTION
**Pull request by Ishad-M-I-M**
## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

File: utils/copy-button-transformer.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      opacity: 1;
+      & .ready { display: none; }
+      & .success { display: block; }
+    }
+`
+    );
+    return trimWhitespace(copyButtonStyle2);
+}
+
+export { transformerCopyButton };
```
Comment (sm1990) [inline utils/copy-button-transformer.js:115]:
```suggestion
export { transformerCopyButton };

```

File: utils/copy-button-transformer.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                            visibility: options.visibility
+                        })
+                    }
+                ]
+            });
+        }
+    };
+}
+function copyButtonStyle({
+                             copyIcon = "data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 48 48'%3E%3Cpath fill='%23adadad' d='M16.187 9.5H12.25a1.75 1.75 0 0 0-1.75 1.75v28.5c0 .967.784 1.75 1.75 1.75h23.5a1.75 1.75 0 0 0 1.75-1.75v-28.5a1.75 1.75 0 0 0-1.75-1.75h-3.937a4.25 4.25 0 0 1-4.063 3h-7.5a4.25 4.25 0 0 1-4.063-3M31.813 7h3.937A4.25 4.25 0 0 1 40 11.25v28.5A4.25 4.25 0 0 1 35.75 44h-23.5A4.25 4.25 0 0 1 8 39.75v-28.5A4.25 4.25 0 0 1 12.25 7h3.937a4.25 4.25 0 0 1 4.063-3h7.5a4.25 4.25 0 0 1 4.063 3M18.5 8.25c0 .966.784 1.75 1.75 1.75h7.5a1.75 1.75 0 1 0 0-3.5h-7.5a1.75 1.75 0 0 0-1.75 1.75'/%3E%3C/svg%3E",
```
Comment (sm1990) [inline utils/copy-button-transformer.js:63]: Shall we use the colors and icons used in http://localhost:3000/learn/get-started/ to maintain consistency and improve accessibility.

**Comment by sm1990** [comment]
@Ishad-M-I-M there are build errors in the following pages. Could you please fix them.
```
> Build error occurred
Error: Export encountered errors on following paths:
	/learn/integration-tools/[...slug]: /learn/integration-tools/openapi-tool
	/spec/[slug]: /spec/graphql
	/use-cases/integration/ballerina-vs-apollo-for-graphql
```
Screenshot from 2024-10-13 11-59-35
Screenshot from 2024-10-13 11-57-55
Screenshot from 2024-10-13 11-51-55

**Comment by Ishad-M-I-M** [comment]
> @Ishad-M-I-M there are build errors in the following pages. Could you please fix them.
> ```
> > Build error occurred
> Error: Export encountered errors on following paths:
> 	/learn/integration-tools/[...slug]: /learn/integration-tools/openapi-tool
> 	/spec/[slug]: /spec/graphql
> 	/use-cases/integration/ballerina-vs-apollo-for-graphql
> ```
> Screenshot from 2024-10-13 11-59-35
> Screenshot from 2024-10-13 11-57-55
> Screenshot from 2024-10-13 11-51-55
> 

Okay. Sure.
I checked most of the pages. May got missed out these. Please let me know if 
there anymore

File: utils/copy-button-transformer.js (hunk 2 above)
Comment (Ishad-M-I-M) [reply utils/copy-button-transformer.js]: Done

**Comment by Ishad-M-I-M** [comment]
@sm1990 Addressed the review comment. Please check

File: utils/copy-button-transformer.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+}) {
+    return {
+        name: "copy-button-transformer",
+        code(node) {
+            node.children.unshift({
+                type: "element",
+                tagName: "button",
+                properties: {
+                    type: "button",
+                    title: "Copy code",
```
Comment (sm1990) [inline utils/copy-button-transformer.js:17]:
```suggestion
                    title: "Copy",
```

File: utils/copy-button-transformer.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                    "aria-label": "Copy code",
```
Comment (sm1990) [inline utils/copy-button-transformer.js:18]:
```suggestion
                    "aria-label": "Copy",
```

Comment (sm1990) [reply utils/copy-button-transformer.js:18]: Can you add the title and the aria-lable to successIcon as `Copied` to match with others.

Comment (Ishad-M-I-M) [reply utils/copy-button-transformer.js]: Done

**Comment by sm1990** [comment]
There is a small styling issue. The icons seems too close to the top margin of the box. Its visible only in http://localhost:3000/use-cases/integration/ballerina-vs-apollo-for-graphql/. I guess its because the box we have used there is different. Apart from that everything else is ok :smile: 
image

**Comment by Ishad-M-I-M** [comment]
@sm1990 fixed the styling issue. Please check it

**Review by sm1990 — APPROVED** [review]

**Comment by sm1990** [comment]
Thank you @Ishad-M-I-M for your contribution. Please follow the steps in https://ballerina.io/hacktoberfest/#rules and fill the form to claim your credits. :smile:

**Pull request merged**